### PR TITLE
Revert the "Blog Home" template name change in the Site Editor to "Home"

### LINKED ISSUE #1262: Home template called "Blog Home"
### Description

In recent versions of Gutenberg/WP, the "Home" Site Editor template is called "Blog Home" -- unless a specific home page has been defined (?)

Given that WordPress is not strictly a blogging CMS, the name is rather unfortunate as the absence of a "Home" page does not mean that any posts will exist, or that the primary purpose of the site will be to publish blog posts. 

A temporary fix would be to patch the name for Woo Express / Woo sites using the `default_template_types` filter.

### PR DESCRIPTION
**Pull request by manospsyx**
### Changes proposed in this Pull Request:

Changes the "Blog Home" template name in the Site Editor back to "Home".

Closes #1262

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

On a fresh Woo Express dev site: 

1. Enable Tsubaki.
2. Go to **Appearance > Editor > Templates**.
3. Double check that the home page template is called "Blog Home".

With the changes in this PR enabled:
1. Go to **Appearance > Editor > Templates**.
2. The home page template should now be called "Home".

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by xristos3490 — APPROVED** [review]
Tested well! Just a heads up, this PR also changes the name under **Appearance > Editor > Pages.**

Tbh, I believe we should address this issue in Gutenberg, as we previously discussed. The use of the “Blog” prefix appears to be overly opinionated.

File: includes/class-wc-calypso-bridge-fse.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+		if ( is_admin() ) {
+
+			// Change "Blog Home" template name in the Site Editor to something more suitable for Woo Express.
+			add_filter(
+				'default_template_types',
+				function( $arg ) {
+
+					if ( isset( $arg[ 'home' ][ 'title' ] ) && _x( 'Blog Home', 'Template name' ) === $arg[ 'home' ][ 'title' ] ) {
+						$arg[ 'home' ][ 'title' ] = __( 'Home' );
```
Comment (xristos3490) [inline includes/class-wc-calypso-bridge-fse.php:53]: Is it safe to omit the translation domain here? Should it be `gutenberg`?

Comment (PanosSynetos) [reply includes/class-wc-calypso-bridge-fse.php:53]: It's ok to use `__( 'Home' )` , as it's the default WP translation

**Review by PanosSynetos — APPROVED** [review]
👌

**Pull request merged**